contents/data/databases.md

```diff
@@ -0,0 +1,115 @@
+
+Take for instance, a program that transfers money from one bank account to another. The actions that the program would need to do involves:
+
+1. Checking if the sender has enough money
+2. Deducting the amount of money to be sent
+3. Adding that amount to the receiver's account
+
+A problem arises if another action takes place in between any of the steps (e.g. another transfer from the same sender), or if any of the steps fails (e.g. due to a program crash). As a result, it may cause the data to be manipulated in an undesired manner (e.g. money not credited to receiver).
+
+Thus, **transactions** allow us to guard against these problems. A **transaction** symbolises a logical unit of work, which consists of multiple database actions, performed on a set of databases. [[Source]](https://en.wikipedia.org/wiki/Database_transaction) Properties of a database transaction ensures that these actions are done in a predictable and reliable manner.
+
+Thus, the transaction that would be implemented in the program could be something like this:
+
+1. Perform funds transfer from *sender* to *receiver*
+    - Checking if the sender has enough money
+    - Deducting the amount of money to be sent
+    - Adding that amount to the receiver's account
+
+As demonstrated above, the actions that needs to be done in order to transfer funds is wrapped as one large **transaction**. Hence, the *funds transfer* can be seen as the unit of work to be done on the database. The actions that make up the **transaction** are executed as though **transaction** is a single action.
+
+### ACID in transactions
+
+For a **transaction** to be considered as an implementation of the **transaction** concept, it has to satisfy the ACID principle.
+
+This principle states that a **transaction** must contain these characteristics:
+- Atomicity - transactions only succeed if all parts of the transaction succeeds. That is to say, if any action fails, the transaction fails and the state of the database should be left unchanged (i.e. as if the transaction did not happen)
+- Consistency - transactions must ensure that the database remains in a valid state after the transaction (for instance, all relations hold true)
+- Isolation - if multiple transactions runs at the same time, the result should be the same as though these transactions are run sequentially
+- Durability - transactions and its results should remain persistent (i.e. power loss or reboot should not affect results)
+
+## Topics for Further Exploration
+
+As this chapter serves to introduce readers to the basics of databases, there are a lot of concepts that are not covered in this chapter.
+
+Now that you know the purpose and characteristics of databases and DBMS, why not go deeper? Here are some topics relating to databases for further exploration by the reader.
+
+- Relational algebra \& calculus
```

Comment:
May be give an brief preview of what these things are and why the reader should learn more about it.